engine-runner-bin/build.rs

```diff
@@ -1,8 +1,56 @@
-use std::env;
+use std::{
+	env,
+	error::Error,
+	fs::File,
+	io::{copy, BufWriter},
+	path::Path,
+};
 
 use engine_upgrade_utils::{
 	build_helpers::toml_with_package_version, ENGINE_LIB_PREFIX, NEW_VERSION, OLD_VERSION,
 };
+use reqwest::blocking::get;
+
+// TODO: Download from mainnet repo if it exists and verify signature.
+// TODO: If we're doing a release build we should force use mainnet binaries. PRO-1622
+fn download_old_dylib(dest_folder: &Path) -> Result<(), Box<dyn Error>> {
+	let target: String = env::var("TARGET").unwrap();
+
+	let prebuilt_supported =
+		target.contains("aarch64-apple-darwin") || target.contains("x86_64-unknown-linux-gnu");
+
+	let shared_lib_ext = if target.contains("apple") { "dylib" } else { "so" };
+
+	let underscored_version = OLD_VERSION.replace('.', "_");
+	let dylib_name = format!("libchainflip_engine_v{underscored_version}.{shared_lib_ext}");
+
+	let dylib_location = dest_folder.join(&dylib_name);
+
+	// If prebuilt is supported we download every time. This is to ensure that if we have retagged,
+	// or added another commit on top then we get the latest build artifacts for a particular
+	// version.
+	if prebuilt_supported {
+		let response = get(format!("https://artifacts.chainflip.io/{OLD_VERSION}/{dylib_name}"))?;
+
+		if response.status().is_success() {
+			std::fs::create_dir_all(dest_folder)?;
+			let mut dest: BufWriter<File> = BufWriter::new(File::create(dylib_location)?);
+			let content = response.bytes()?;
+			copy(&mut content.as_ref(), &mut dest)?;
```

Comment:
Looks like this can be written directly using response.copy_to(..) to avoid allocating the additional `content` Vec (which is quite big?).